datafusion/src/physical_plan/cross_join.rs

```diff
@@ -331,8 +331,7 @@ fn build_batch(
             let scalar = ScalarValue::try_from_array(arr, left_index)?;
             Ok(scalar.to_array_of_size(batch.num_rows()))
         })
-        .collect::<Result<Vec<_>>>()
-        .map_err(|x| x.into_arrow_external_error())?;
+        .collect::<Result<Vec<_>>>()?;
```

Comment:
this is a pretty good exemplar of how this change makes the DataFusion codebase less confusing